src/components/DataTable.vue

```diff
@@ -1,160 +1,315 @@
+      <p class="data-table__info">
         Showing {{ first + 1 }} to {{ last }} of {{ totalRecords }}
-      </span>
-    </section>
-  </template>
-  
-  <script>
-  import TreeTable from 'primevue/treetable';
-  import Column from 'primevue/column';
-  import IconField from 'primevue/iconfield';
-  import InputIcon from 'primevue/inputicon';
-  import InputText from 'primevue/inputtext';
-  
-  export default {
-    components: { TreeTable, Column, IconField, InputIcon, InputText },
-    props: {
-      nodes: {
-        type: Object,
-        required: true,
-        default: () => ({}),
-      },
-      rows: {
-        type: Number,
-        default: 10,
-      },
-      loading: {
-        type: Boolean,
-        default: false,
-      },
-      totalRecords: {
-        type: Number,
-        default: 0,
-      },
-      filters: {
-        type: Object,
-        default: () => ({}),
-      },
-      first: {
-       type: Number,
-       default: 0,
-     },
-     last: {
-       type: Number,
-       default: 0,
-     },
-    },
-    emits: ['update:page', 'update:filters'],
-    methods: {
-      onPage(event) {
-        this.$emit('update:page', event);
-      },
-      onFilterChange(event) {
-        this.$emit('update:filters', event)
-      }
-   },
+      </p>
+      <div class="data-table__pagination">
+        <button :disabled="currentPage === 1" class="pagination-button" @click="prevPage">
+          <i class="pi pi-chevron-left" style="font-size: 0.6rem;"></i>
+        </button>
+        <div 
+          v-for="page in totalPages"
+          :key="page"
+          :class="{ page_selected: page === currentPage}" 
+          class="pagination-page"
+          @click="onPage(page)"
+        >
+          {{ page }}
+        </div>
+        <button :disabled="currentPage === totalPages" class="pagination-button" @click="nextPage">
+          <i class="pi pi-chevron-right" style="font-size: 0.6rem;"></i>
+        </button>
+      </div>
+    </div>
+  </section>
+</template>
+
+<script>
+export default {
+name: "AppTable",
+props: {
+  rowsPerPage: {
+    type: Number,
+    default: 10,
+  },
+  data: {
+      type: Object,
+      required: true,
+      default: () => ({}),
+  },
+  totalRecords: {
+    type: Number,
+    default: 0,
+  },
+  headers: {
+    type: Array,
+    default: () => {[]}
+  }
+},
+data() {
+  return {
+    searchQuery: '',
+    currentPage: 1,
+    sortField: null,
+    sortOrder: 1,
+    first: 0,
+    last: 10,
   };
-  </script>
-<style lang="scss">
-.tree-table__wrapper {
-  position: relative;
+},
+computed: {
+  filteredData() {
+    if (!this.searchQuery) return this.data;
+
+    return this.data.filter((item) =>
+      Object.values(item).some((value) =>
+        String(value).toLowerCase().includes(this.searchQuery.toLowerCase())
+      )
+    );
+  },
+  sortedData() {
+    if (!this.sortField) return this.filteredData;
+
+    return [...this.filteredData].toSorted((a, b) => {
```

Comment:
you don't have to use spread operator here to create a clone, because `toSorted` returns a new array. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted

Reply:
Thanks for noticing. I had used the sort method before, and after making the change, I just forgot to remove it